### PR TITLE
fix analyzer deprecations

### PR DESCRIPTION
**Pull request by jakemac53**
cc @scheglov do I need to up these min sdk constraints or should I leave them?

### CODE REVIEW

**Comment by scheglov** [comment]
Not sure why you asking me, you probably know this stuff better :-). For SDK constraints, we probably need to say that it is `< 3.0.0`. But we don't have to say `>= 2.17.0` because at least the analyzer will require this, so I think you will not able to satisfy constraints on lower SDK versions anyway.

For `analyzer` constraints, `analyzer: '>=5.1.0 <6.0.0'` seems right to me, because you need `>=5.1.0` to get new incarnation of `get name`.

**Comment by jakemac53** [comment]
I think I need to update the sdk dependency to >=2.18.0 across the board here so we can get a package:test version that supports the latest analyzer

**Comment by jakemac53** [comment]
@scheglov sorry I meant the min analyzer constraints, just had a brain fart :)

Thanks you answered my question for that, I will leave it at `>=5.1.0`

**Review by natebosch — APPROVED** [review]

File: build_modules/mono_pkg.yaml [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-- analyze_and_format:
-  - group:
-    - format
-    - analyze: --fatal-infos .
-- unit_test:
-  - test: -P presubmit --test-randomize-ordering-seed=random
-    os:
-    - linux
-    - windows
+  - analyze_and_format:
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply build_modules/mono_pkg.yaml:6]: Oh I think my yaml formatter did that, fwiw I do prefer this style but I could try and change my settings if we want it the other way

[the remark this answers is not in the document]
Comment (natebosch) [reply build_modules/mono_pkg.yaml:6]: The other way matches the default yaml indentation in my editor, and it matches what I see in all the yaml docs I checked in google3. We don't have an official style guide though, and I'm noticing that google's public facing docs which use yaml are not consistent at all. Without a shared formatter I don't know if it matters much.

If your formatter is configurable, I think we should aim for not-indented lists. If it isn't, 🤷 it's fine to have some churn in the lines.

[the remark this answers is not in the document]
Comment (jakemac53) [reply build_modules/mono_pkg.yaml:6]: I don't see it being configurable, but I can disable it entirely, I mostly have it for the schema validation

[the remark this answers is not in the document]
Comment (jakemac53) [reply build_modules/mono_pkg.yaml]: I went ahead and just reverted those changes and disabled my formatter

**Pull request merged**